Userland/Applications/FileManager/DirectoryView.cpp

```diff
@@ -406,18 +405,19 @@ void DirectoryView::add_path_to_history(DeprecatedString path)
 
 bool DirectoryView::open(DeprecatedString const& path)
 {
-    auto real_path = Core::DeprecatedFile::real_path_for(path);
-    if (real_path.is_null() || !FileSystem::is_directory(path))
+    auto real_path_or_error = FileSystem::real_path(path);
+    if (real_path_or_error.is_error() || !FileSystem::is_directory(path))
         return false;
 
-    if (chdir(real_path.characters()) < 0) {
+    auto real_path = real_path_or_error.release_value();
+    if (Core::System::chdir(real_path).is_error())
         perror("chdir");
```

Comment:
Pre-existing bug, but `perror` only prints an error - we also need to return false here!